docs/source/developers/java/building.rst

```diff
@@ -0,0 +1,308 @@
+    [INFO] Arrow Algorithms ................................... SUCCESS [ 32.442 s]
+    [INFO] Arrow Performance Benchmarks ....................... SUCCESS [  5.418 s]
+    [INFO] Arrow Java C Data Interface ........................ SUCCESS [  7.907 s]
+    [INFO] ------------------------------------------------------------------------
+    [INFO] BUILD SUCCESS
+    [INFO] ------------------------------------------------------------------------
+
+Build Java C++ Interface
+########################
+
+Java arrow-jni maven profile (build ORC / Gandiva / Dataset):
+
+.. code-block::
+
+    $ cd arrow/java
+    $ mvn -Darrow.cpp.build.dir=$GITHUB_WORKSPACE/java-dist/lib -Parrow-jni clean install
+    [INFO] Reactor Summary for Apache Arrow Java Root POM 7.0.0:
+    [INFO]
+    [INFO] Apache Arrow Java Root POM ......................... SUCCESS [  7.342 s]
+    [INFO] Arrow Format ....................................... SUCCESS [  2.417 s]
+    [INFO] Arrow Memory ....................................... SUCCESS [  1.967 s]
+    [INFO] Arrow Memory - Core ................................ SUCCESS [  4.714 s]
+    [INFO] Arrow Memory - Unsafe .............................. SUCCESS [  3.157 s]
+    [INFO] Arrow Memory - Netty ............................... SUCCESS [  3.334 s]
+    [INFO] Arrow Vectors ...................................... SUCCESS [ 21.791 s]
+    [INFO] Arrow Compression .................................. SUCCESS [  3.854 s]
+    [INFO] Arrow Tools ........................................ SUCCESS [  8.359 s]
+    [INFO] Arrow JDBC Adapter ................................. SUCCESS [  8.847 s]
+    [INFO] Arrow Plasma Client ................................ SUCCESS [  2.459 s]
+    [INFO] Arrow Flight ....................................... SUCCESS [  2.357 s]
+    [INFO] Arrow Flight Core .................................. SUCCESS [ 38.837 s]
+    [INFO] Arrow Flight GRPC .................................. SUCCESS [  5.955 s]
+    [INFO] Arrow Flight SQL ................................... SUCCESS [ 17.390 s]
+    [INFO] Arrow Flight Integration Tests ..................... SUCCESS [  6.148 s]
+    [INFO] Arrow AVRO Adapter ................................. SUCCESS [  9.635 s]
+    [INFO] Arrow Algorithms ................................... SUCCESS [ 26.949 s]
+    [INFO] Arrow Performance Benchmarks ....................... SUCCESS [  4.211 s]
+    [INFO] Arrow Orc Adapter .................................. SUCCESS [  6.522 s]
+    [INFO] Arrow Gandiva ...................................... SUCCESS [01:20 min]
+    [INFO] Arrow Java Dataset ................................. SUCCESS [ 12.949 s]
+    [INFO] ------------------------------------------------------------------------
+    [INFO] BUILD SUCCESS
+    [INFO] ------------------------------------------------------------------------
+
+IDE Configuration
+=================
+
+Go to open java project and select java folder.
```

Comment:
Add a section so it's clear this is for IntelliJ

Reply:
Added